configs/seL4Config.cmake

```diff
@@ -244,16 +251,6 @@ if(
     configure_file(
         "${CMAKE_CURRENT_LIST_DIR}/../${toolchain_file}" "${toolchain_outputfile}.temp" @ONLY
     )
-    if(EXISTS "${toolchain_outputfile}")
```

Comment:
@kent-mcleod: what do you think about removing this check? I assume it there for a reason mainly coming from using the CMake GUI to change the settings. So we might better keep this to avoid broken builds then.

Reply:
I edit CMakeCache.txt directly. For some reason this don't allow me to compile.

To avoid broken builds, it's better to add protection in CMake scripts directly.

Reply:
It sounds to me like the check did exactly what it was intended to do. Editing `CMakeCache.txt` directly is not supported by the rest of the build system and will lead to inconsistent results.

Reply:
> It sounds to me like the check did exactly what it was intended to do. Editing `CMakeCache.txt` directly is not supported by the rest of the build system and will lead to inconsistent results.

The error should have only been raised if the changes to the CMakeCache.txt changed the contents of the generated toolchain file. It shouldn't raise an error for simply editing the CMakeCache.txt file as this is what the `ccmake` program essentially does - which is supported.  However making config changes that affect the generated toolchain file (which should be restricted to changes in the following cache variables: `KernelSel4Arch`, `KernelArch`, `KernelWordSize`, `CROSS_COMPILER_PREFIX`) will generate this error as changing the toolchain file requires re-initializing the build because CMake caches a lot of toolchain discovery it does and this doesn't get reset if the toolchain file changes.